test.js

```diff
@@ -158,6 +154,27 @@ describe('capture-exit', function() {
         exit.onExit(function () { });
       }).to.throw('Cannot install handler when exit is not captured.  Call `captureExit()` first');
     });
+
+    it('throws if an exit is already happening', function() {
```

Comment:
Ah, this isn't self imposed, rather dictated by nodes requirement of sync only handles in this position